packages/did/src/DidDetails/DidDetails.ts

```diff
@@ -75,16 +83,35 @@ export abstract class DidDetails implements IDidDetails {
     return firstAuthenticationKey
   }
 
-  public get encryptionKey(): DidKey | undefined {
-    return this.getKeys(KeyRelationship.keyAgreement)[0]
+  /**
+   * Returns the first encryption key of the DID, if any.
+   *
+   * @returns The first encryption key, in the order they are stored internally, of the given DID.
+   */
+  public get encryptionKey(): DidEncryptionKey | undefined {
+    return this.getKeys(KeyRelationship.keyAgreement)[0] as DidEncryptionKey
```

Comment:
C: more `as` in this file

Reply:
https://github.com/KILTprotocol/sdk-js/pull/473/commits/5234b12f7bc5848b609e38cd75eea91aaf4ffd2e